src/components/Fieldset/Fieldset.jsx

```diff
@@ -0,0 +1,184 @@
+						placeholder={placeholder}
+						onBlur={onBlur}
+						onChange={onChange}
+						onClick={onClick}
+						onFocus={onFocus}
+						onInvalid={onInvalid}
+						readOnly={readOnly}
+						ref={inputRef}
+						required={required}
+					/>
+				}
+
+				{!isMultiline &&
+					<input
+						className={cx( 'input', { ...sharedClassNames } )}
+						type={type}
+						aria-describedby={describedBy}
+						autocomplete={autoComplete}
+						defaultValue={defaultValue}
+						describedBy={describedBy}
+						disabled={disabled}
+						id={id}
+						maxLength={maxLength}
+						placeholder={placeholder}
+						readOnly={readOnly}
+						onBlur={onBlur}
+						onChange={onChange}
+						onClick={onClick}
+						onFocus={onFocus}
+						onInvalid={onInvalid}
+						pattern={pattern}
+						readOnly={readOnly}
+						ref={inputRef}
+						required={required}
+						type={type}
+					/>
+				}
+			</label>
+		</fieldset>
+	);
+};
+
+Fieldset.propTypes = {
+	/**
+	 * Input tag prop; boolean to accept autoCompletion or not.
+	 */
+	autoComplete: PropTypes.string,
+	/**
+	 * Input tag prop; sets default value.
+	 */
+	defaultValue: PropTypes.string,
+	/**
+	 * Input tag prop; used with aria-describedby attribute to indicate the ID of a linked description element.
+	 */
+	describedBy: PropTypes.string,
+	/**
+	 * Input tag prop; sets whether the input is disabled from changing values or not.
+	 */
+	disabled: PropTypes.bool,
+	/**
+	 * Used to identify the element.
+	 */
+	id: PropTypes.string,
+	/**
+	 * Used in the event of forwarding a ref to the input tag.
+	 */
+	inputRef: PropTypes.object,
+	/**
+	 * Boolean that determines whether component is a `textarea` or `input` tag.
+	 */
+	isMultiline: PropTypes.bool.isRequired,
+	/**
+	 * Text to be used in `label` tag.
+	 */
+	label: PropTypes.string,
+	/**
+	 * Input tag prop; prevents string length beyond given value.
+	 */
+	maxLength: PropTypes.number,
+	/**
+	 * Event handler for blur event.
+	 */
+	onBlur: PropTypes.func.isRequired,
+	/**
+	 * Event handler for change event.
+	 */
+	onChange: PropTypes.func.isRequired,
+	/**
+	 * Event handler for click event.
+	 */
+	onClick: PropTypes.func,
+	/**
+	 * Event handler for focus event.
+	 */
+	onFocus: PropTypes.func.isRequired,
+	/**
+	 * Event handler for invalid event.
+	 */
+	onInvalid: PropTypes.func,
+	/**
+	 * Input tag prop; sets `value` validation pattern.
+	 */
+	pattern: PropTypes.string,
+	/**
+	 * Input tag prop; sets 'ghost' text that appears when no value exists.
+	 */
+	placeholder: PropTypes.string,
+	/**
+	 * Input tag prop; determines whether value is editable or not.
+	 */
+	readOnly: PropTypes.bool,
+	/**
+	 * Input tag prop + style prop; if true, value must be present for the form to be submitted. Will add asterisk to label if label is present.
+	 */
+	required: PropTypes.bool.isRequired,
+	/**
+	 * Input tag prop; determines type of form control.
+	 */
+	type: PropTypes.oneOf( [ 'text', 'email', 'password' ] ),
```

Comment:
there are a few more here, off the top of my head there's `tel` and `search`, possibly more

Reply:
also include in the description that this will apply to non-multiline only